javaagent-core/src/main/java/org/hypertrace/agent/core/config/HypertraceConfig.java

```diff
@@ -236,4 +258,40 @@ private static String convertYamlToJson(InputStream yaml) throws IOException {
     ObjectMapper jsonWriter = new ObjectMapper();
     return jsonWriter.writeValueAsString(obj);
   }
+
+  public static class DynamicConfigFetcher implements Runnable {
+
+    private final ConfigurationServiceGrpc.ConfigurationServiceBlockingStub blockingStub;
+
+    private static Int64Value configTimestamp;
+
+    private DynamicConfigFetcher(String dynamicConfigServiceUrl) {
+      Channel channel = ManagedChannelBuilder.forTarget(dynamicConfigServiceUrl).build();
+      blockingStub = ConfigurationServiceGrpc.newBlockingStub(channel);
+      configTimestamp = Int64Value.newBuilder().setValue(System.currentTimeMillis()).build();
+    }
+
+    @Override
+    public void run() {
+      Service.UpdateConfigurationResponse response =
+          blockingStub.updateConfiguration(
+              Service.UpdateConfigurationRequest.newBuilder()
+                  .setTimestamp(configTimestamp)
+                  .build());
+      configTimestamp = response.getTimestamp();
+      AgentConfig.Builder configBuilder = HypertraceConfig.get().toBuilder();
+      configBuilder.setEnabled(response.getEnabled());
+      configBuilder.setDataCapture(response.getDataCapture());
+      configBuilder.setJavaagent(response.getJavaAgent());
+      agentConfig = configBuilder.build();
```

Comment:
is this the field from the HypertraceConfig class?

Reply:
Should we worry about concurrent modifications (read/write) at the same time?

Reply:
Yes, this is the field from HypertraceConfig class. We do not need to worry about concurrent modifications as it is a volatile field.